org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -183,15 +196,45 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 				if (source == null) {
 					if (status == null) {
 						// did not find source
-						MessageDialog.openInformation(JDIDebugUIPlugin.getActiveWorkbenchShell(), ConsoleMessages.JavaStackTraceHyperlink_Information_1, NLS.bind(ConsoleMessages.JavaStackTraceHyperlink_Source_not_found_for__0__2, new String[] {typeName}));
+						MessageDialog.openInformation(JDIDebugUIPlugin.getActiveWorkbenchShell(), ConsoleMessages.JavaStackTraceHyperlink_Information_1, NLS.bind(ConsoleMessages.JavaStackTraceHyperlink_Source_not_found_for__0__2, new String[] {
+								typeName }));
 					} else {
 						JDIDebugUIPlugin.statusDialog(ConsoleMessages.JavaStackTraceHyperlink_3, status);
 					}
 				} else if (source instanceof List) { // ambiguous
 					@SuppressWarnings("unchecked")
 					List<Object> matches = (List<Object>) source;
+					List<Object> exactMatchesFiltered = new ArrayList<>();
+					String originalHyperLink2 = originalHyperLink;
+					int firstMethodStartIndex = originalHyperLink2.indexOf('.');
+					int firstMethodClosing = originalHyperLink2.indexOf(')');
+					if (firstMethodStartIndex != -1 && firstMethodClosing != -1) {
+						String firstMethod = originalHyperLink2.substring(firstMethodStartIndex + 1, firstMethodClosing + 1);
+						for (Object obj : matches) {
+							@SuppressWarnings("restriction")
+							SourceType source = (SourceType) obj;
```

Comment:
use `IType `and do an instanceof check to prevent errors

Reply:
Okay, will do this change right now

Reply:
Changes done and tested now.